13/add subscribe page

AdamVig opened the pull request:
Closes #13.

![stockpile-subscribe-v1](https://user-images.githubusercontent.com/5354752/28740836-45d634de-73d8-11e7-82fa-fc6cb5d82f32.gif)

AdamVig committed:
Add subscribe page

AdamVig committed:
Add Browserify to build script

AdamVig committed:
Change password strength test to Node module

AdamVig committed:
Disable removing input type=text attribute

The HTML minifier was removing this attribute because it is technically redundant, but its removal was causing input styling to break.

AdamVig committed:
Fix linter errors

AdamVig committed:
Add form autocomplete sections

AdamVig committed:
Finish building subscribe page

AdamVig committed:
Fix indentation

AdamVig committed:
Move instructions to form section

This prevents them from being shown after successful subscription, which would be redundant.

AdamVig committed:
Fix content being too wide for viewport

The bug was caused by margins on the Flexboxgrid rows. Removing those margins fixes it, and removes the need for hiding overflow on the `body` and `html` elements.

- remove Normalize
- add global styles and include in every page

AdamVig committed:
Improve layout

- move instructions into the form and rephrase to be more helpful
- remove reundant styles
- add padding to form on small screens
- make form width fit on small screens

emroussel commented on src/scripts/subscribe.js:
```diff
@@ -0,0 +1,124 @@
+  owaspPasswordStrengthTest.config({
+    allowPassphrases: true,
+    maxLength: 128,
+    minLength: 10,
```
Any reason why you chose 10 for the min password length? It does enforce stronger password, but I wonder if the number is a little too high that it might cause frustration for the users. Most websites I looked at (Google, GitHub, etc.) use 7 or 8.

AdamVig replied:
Good point, this is too strict of a requirement. The [NIST password guidelines](https://pages.nist.gov/800-63-3/sp800-63b.html#5111-memorized-secret-authenticators) recommend a minimum length of 8 characters, so I'll change it to that.

emroussel commented:
Also, have you thought about adding immediate feedback for the fields (doing the checks for having something in each field + a valid email and password as the user types instead of waiting until he or she clicks on subscribe)? I feel like it would provide a better UX.

AdamVig commented:
You're right, it should validate the fields as the user goes through the form, instead of after they attempt to submit. [Validation should be either on blur or on submit](https://ux.stackexchange.com/questions/48268/validate-on-blur-or-keypress), and I lean towards on blur, so I'll implement it that way.

AdamVig committed:
Reduce minimum password length to 8 characters

This is the minimum length recommended by NIST.

AdamVig committed:
Fix layout on large screens

Adding the `container` class to a row messed up the margins, causing the row to be offset to the left side instead of centered.

AdamVig committed:
Validate all fields on blur
